cluster-autoscaler/core/scale_down.go

```diff
@@ -1065,7 +1065,34 @@ func (sd *ScaleDown) scheduleDeleteEmptyNodes(emptyNodes []*apiv1.Node, client k
 			return deletedNodes, errors.ToAutoscalerError(errors.ApiCallError, taintErr)
 		}
 		deletedNodes = append(deletedNodes, node)
-		go func(nodeToDelete *apiv1.Node, nodeGroupForDeletedNode cloudprovider.NodeGroup) {
+		go func(nodeToDelete *apiv1.Node, nodeGroupForDeletedNode cloudprovider.NodeGroup, daemonSetEvictionForEmptyNodes bool) {
+			if daemonSetEvictionForEmptyNodes {
+				nodeInfo, _ := sd.context.ClusterSnapshot.NodeInfos().Get(nodeToDelete.Name)
+				_, daemonSetPods, _, _ := simulator.FastGetPodsToMove(nodeInfo, true, true, []*policyv1.PodDisruptionBudget{})
+
+				dsEviction := make(chan status.PodEvictionResult, len(daemonSetPods))
+
+				// Perform eviction of DaemonSet pods. We don't want to raise an error if daemonsetPod wasn't evict properly
+				retryUntil := time.Now().Add(10 * time.Second)
+				for _, daemonSetPod := range daemonSetPods {
+					go func(podToEvict *apiv1.Pod) {
+						dsEviction <- evictPod(podToEvict, true, client, recorder, sd.context.MaxGracefulTerminationSec, retryUntil, 5*time.Second)
+					}(daemonSetPod)
+				}
+
+				// wait for creating eviction of DaemonSet pods
+				waitForDsEviction := time.Now().Add(10 * time.Second)
+				counter := 0
+				for range daemonSetPods {
+					select {
+					case <-dsEviction:
+						counter++
+					case <-time.After(waitForDsEviction.Sub(time.Now()) + 5*time.Second):
```

Comment:
You're specifying a deadline (`time.Now().Add(10 * time.Second)`), and then substracting `time.Now()` from that deadline to get a timeout value - seems redundant. Also, this value seems repeated from `retryUntil`.

WDYT about defining `dsEvictionTimeout := 10*time.Second` and `retryUntil := time.Now().Add(dsEvictionTimeout)` at the top, and passing `dsEvictionTimeout + 5*time.Second` to `time.After()`?